Sources/Tracking/CustomerIOImplementation+ScreenViews.swift

```diff
@@ -0,0 +1,58 @@
+import Foundation
+#if canImport(UIKit)
+import UIKit
+
+public extension CustomerIOImplementation {
+    func enableAutoScreenviewTracking() {
+        let selector1 = #selector(UIViewController.viewDidAppear(_:))
+        let selector2 = #selector(CustomerIOImplementation._swizzled_UIKit_viewDidAppear(_:))
+        guard let originalMethod = class_getInstanceMethod(UIViewController.self, selector1) else {
+            return
+        }
+        guard let swizzleMethod = class_getInstanceMethod(CustomerIOImplementation.self, selector2) else {
+            return
+        }
+        method_exchangeImplementations(originalMethod, swizzleMethod)
+    }
+
+    @objc dynamic func _swizzled_UIKit_viewDidAppear(_ animated: Bool) {
+        _swizzled_UIKit_viewDidAppear(animated)
+
+        guard let delegate = UIApplication.shared.delegate else {
+            return
+        }
+
+        guard let window = delegate.window else {
+            return
+        }
+
+        var viewController = window!.rootViewController
+        if viewController is UINavigationController {
+            viewController = (viewController as! UINavigationController).visibleViewController
+        }
+
+        var name = String(describing: type(of: viewController)).replacingOccurrences(of: "ViewController", with: "")
+
+        if name.isEmpty || name == "" {
+            if viewController?.title != nil {
+                name = viewController?.title ?? "Unknown"
```

Comment:
default value will never be executed since the check is for `viewController?.title != nil` which means the code block will be executed only when value is not nil. Also, it is always better to use if let in such cases